quickannotator/db/crud/annotation.py

```diff
@@ -122,22 +168,63 @@ def insert_annotations(self, polygons: List[BaseGeometry], tile_ids: List[int] |
 
     # READ
     def get_annotation_by_id(self, annotation_id: int) -> db_models.Annotation:
-        result = get_annotation_query(self.model, 1/self.scaling_factor).filter_by(id=annotation_id).first()
+        result = self.query.filter_by(id=annotation_id).first()
         return result
 
+
     def get_annotations_by_ids(self, annotation_ids: List[int]) -> List[db_models.Annotation]:
-        result = get_annotation_query(self.model, 1/self.scaling_factor).filter(self.model.id.in_(annotation_ids)).all()
+        result = self.query.filter(self.model.id.in_(annotation_ids)).all()
         return result
 
+
     def get_annotations_for_tiles(self, tile_ids: List[int]) -> List[db_models.Annotation]:
-        result = get_annotation_query(self.model, 1/self.scaling_factor).filter(self.model.tile_id.in_(tile_ids)).all()
+        result = self.query.filter(self.model.tile_id.in_(tile_ids)).all()
+        return result
+    
+
+    def get_all_annotations(self) -> List[models.Annotation]:
+        result = self.query.all()
         return result
 
+
     def get_annotations_within_poly(self, polygon: BaseGeometry) -> List[db_models.Annotation]:
         scaled_polygon = self.scale_polygon(polygon, self.scaling_factor)
         # NOTE: Sqlite may not use the spatial index here.
-        result = get_annotation_query(self.model, 1/self.scaling_factor).filter(func.ST_Intersects(self.model.polygon, func.ST_GeomFromText(scaled_polygon.wkt, 0))).all()
+        result = self.query.filter(func.ST_Intersects(self.model.polygon, func.ST_GeomFromText(scaled_polygon.wkt, 0))).all()
         return result
+    
+
+    def export_all_annotations_to_tar(self, tarpath: str):
+        """
+        Saves all annotations to a tar archive.
+
+        Args:
+            tarname (str): The name of the tar file to save the annotations to.
+        """
+        with tarfile.open(tarpath, mode='w') as tar:
+            annotations = self.get_all_annotations()
+            feature_collection = anns_to_feature_collection(annotations)
+            feature_collection_json = geojson.dumps(feature_collection)
+
+            # Create a tarinfo object for the GeoJSON file
+            tarinfo = tarfile.TarInfo(name=f"{self.get_annotation_table_name()}.geojson")
+            tarinfo.size = len(feature_collection_json)
```

Comment:
i don't think this is correct, because the actual len/size will depend on the coding used, so this is more correct:

len(feature_collection_json.encode('utf-8'))

however, if you do that, you'll be doing it twice, here and again on line 214, so this should be done once and then reused in both places

Reply:
done.